From d9fb46c2caedf135c21c983a8e379e9bdfc5043d Mon Sep 17 00:00:00 2001
From: Giovanni Campagna <scampa.giovanni@gmail.com>
Date: Wed, 21 Aug 2013 10:02:43 +0200
Subject: [PATCH 2/2] xwayland: shortcut Enter/ExitVT

Avoids a warning due to drmGetMaster and a crash with multimonitor,
caused by not having an intel_mode.
---
 src/uxa/intel_driver.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/uxa/intel_driver.c b/src/uxa/intel_driver.c
index e74ba60..1c38075 100644
--- a/src/uxa/intel_driver.c
+++ b/src/uxa/intel_driver.c
@@ -1175,6 +1175,9 @@ static void I830LeaveVT(VT_FUNC_ARGS_DECL)
 	intel_screen_private *intel = intel_get_screen_private(scrn);
 	int ret;
 
+	if (xorgWayland)
+	  return TRUE;
+
 	xf86RotateFreeShadow(scrn);
 
 	xf86_hide_cursors(scrn);
@@ -1194,6 +1197,9 @@ static Bool I830EnterVT(VT_FUNC_ARGS_DECL)
 	intel_screen_private *intel = intel_get_screen_private(scrn);
 	int ret;
 
+	if (xorgWayland)
+	  return TRUE;
+
 	ret = drmSetMaster(intel->drmSubFD);
 	if (ret) {
 		xf86DrvMsg(scrn->scrnIndex, X_WARNING,
-- 
1.8.3.1

